sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregator.java

```diff
@@ -0,0 +1,166 @@
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.common.InstrumentDescriptor;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.DoubleHistogramData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+import javax.annotation.concurrent.GuardedBy;
+
+final class DoubleHistogramAggregator extends AbstractAggregator<HistogramAccumulation> {
+  private static final long[] countsOfOne = new long[] {1};
+
+  private final double[] boundaries;
+
+  DoubleHistogramAggregator(
+      Resource resource,
+      InstrumentationLibraryInfo instrumentationLibraryInfo,
+      InstrumentDescriptor instrumentDescriptor,
+      double[] boundaries,
+      boolean stateful) {
+    super(resource, instrumentationLibraryInfo, instrumentDescriptor, stateful);
+    this.boundaries = boundaries;
+  }
+
+  @Override
+  public AggregatorHandle<HistogramAccumulation> createHandle() {
+    return new Handle(this.boundaries);
+  }
+
+  /**
+   * Return the result of the merge of two histogram accumulations. As long as one Aggregator
+   * instance produces all Accumulations with constant boundaries we don't need to worry about
+   * merging accumulations with different boundaries.
+   */
+  @Override
+  public final HistogramAccumulation merge(HistogramAccumulation x, HistogramAccumulation y) {
+    long[] mergedCounts = new long[x.getCounts().length];
+    for (int i = 0; i < x.getCounts().length; ++i) {
+      mergedCounts[i] = x.getCounts()[i] + y.getCounts()[i];
+    }
+    return HistogramAccumulation.create(x.getSum() + y.getSum(), mergedCounts);
+  }
+
+  @Override
+  public final MetricData toMetricData(
+      Map<Labels, HistogramAccumulation> accumulationByLabels,
+      long startEpochNanos,
+      long lastCollectionEpoch,
+      long epochNanos) {
+    List<Double> boundaries = new ArrayList<>(this.boundaries.length);
+    for (double v : this.boundaries) {
+      boundaries.add(v);
+    }
+    return MetricData.createDoubleHistogram(
+        getResource(),
+        getInstrumentationLibraryInfo(),
+        getInstrumentDescriptor().getName(),
+        getInstrumentDescriptor().getDescription(),
+        getInstrumentDescriptor().getUnit(),
+        DoubleHistogramData.create(
+            isStateful() ? AggregationTemporality.CUMULATIVE : AggregationTemporality.DELTA,
+            MetricDataUtils.toDoubleHistogramPointList(
+                accumulationByLabels,
+                isStateful() ? startEpochNanos : lastCollectionEpoch,
+                epochNanos,
+                boundaries)));
+  }
+
+  @Override
+  public HistogramAccumulation accumulateDouble(double value) {
+    return HistogramAccumulation.create(value, countsOfOne);
```

Comment:
the decision was an arbitrary one because I'm not sure if histograms would work with AsyncInstruments/Observer and there are not any builders for that if I remember correctly. 

> I mean if the boundaries are {1, 3} and a record of 2 then we need {0, 1, 0} counts correct?

I think this is a better default. but do we really need to implement `accumulateDouble`/`accumulateLong` at all?

Reply:
That is a good question, but probably will need a small design improvement to allow Aggregator to implement/support only the sync instruments.

I think before solving that we should do the correct thing.

Reply:
sounds good. updated.